scripts/o3de/o3de/validation.py

```diff
@@ -83,37 +95,47 @@ def valid_o3de_project_json(file_name: str or pathlib.Path, generate_uuid: bool
     return True
 
 
+def valid_o3de_gem_json_data(json_data: dict) -> bool:
+    try:
+        _ = json_data['gem_name']
+
+        if 'compatible_engines' in json_data:
+            if not utils.validate_version_specifier_list(json_data['compatible_engines']):
+                return False
+
+    except (json.JSONDecodeError, KeyError):
+        return False
+    return True
+
 def valid_o3de_gem_json(file_name: str or pathlib.Path) -> bool:
     file_name = pathlib.Path(file_name).resolve()
     if not file_name.is_file():
         return False
 
     with file_name.open('r') as f:
         try:
-            json_data = json.load(f)
-            _ = json_data['gem_name']
+            return valid_o3de_gem_json_data(json.load(f))
+        except json.JSONDecodeError:
+            return False
 
-            if 'compatible_engines' in json_data:
-                if not utils.validate_version_specifier_list(json_data['compatible_engines']):
-                    return False
 
-        except (json.JSONDecodeError, KeyError):
-            return False
+def valid_o3de_template_json_data(json_data: dict) -> bool:
+    try:
+        _ = json_data['template_name']
+    except (json.JSONDecodeError, KeyError):
```

Comment:
```suggestion
    except (KeyError):
```

Reply:
sometimes I wonder if we should be using 4 lines of code to check if a field is set when we could just use
```
return json_data.get('template_name') is not None
```
though we should probably require the template name has an actual value i.e.
```
return json_data.get('template_name','') != '' 
```